crates/factors-test/src/lib.rs

```diff
@@ -100,6 +101,8 @@ impl<T: RuntimeFactors> TestEnvironment<T> {
 pub async fn build_locked_app(manifest: &toml::Table) -> anyhow::Result<LockedApp> {
     let toml_str = toml::to_string(manifest).context("failed serializing manifest")?;
     let dir = tempfile::tempdir().context("failed creating tempdir")?;
+    // `foo` variable is set to require. As we're not providing a default value, env is checked.
+    std::env::set_var(env_key(None, "foo"), "baz");
```

Comment:
I still say we should be able to transform a TOML manifest into a LockedApp without doing validation on variables.